static/js/components/dashboard/courses/Grades.js

```diff
@@ -57,7 +56,8 @@ const renderFinalGrade = R.ifElse(
   R.compose(
     reduceM("--", renderGrade("Final Grade")),
     S.map(formatGrade),
-    calculateFinalGrade
+    S.filter(R.complement(R.equals(""))),
```

Comment:
I think this can be replaced with `filterEmpty` in `static/js/lib/grades.js`

Reply:
@aliceriot recommended to use `R.equals("")` to be more explicit for what we are checking here.